### PR TITLE
Feature/167 enhancement in user custom field

### PR DESCRIPTION
**Pull request by GauravGusain98**
#167 

### Types of changes

- [ ] Bug fix (a change which fixes an issue)
- [x] New feature (a change which adds functionality)

### Short description of what this resolves:

- Added status column in the user_custom_fields table
- Mark all user custom fields as `inactive` and after processing the webhook data, mark the custom fields active whose values are present.

### Checklist:

- [x] I have performed a self-review of my own code.
- [x] The code follows the style guidelines of this project.
- [x] The code changes are passing the CI checks
- [ ] I have documented my code wherever required.
- [ ] The changes requires a change to the documentation.
- [ ] I have updated the documentation based on the my changes.

### CODE REVIEW

File: api/models/user_custom_fields.py [hunk 1]
```diff
@@ -8,14 +8,38 @@ class UserCustomFieldsQuery(BaseQuery):
     def get_by_user_phone(self, phone):
         return self.filter(UserCustomFields.user_phone == phone).all()
 
+    def mark_user_groups_as_inactive(self, phone):
```
Comment (Satendra-SR) [inline api/models/user_custom_fields.py:11]: Why we are using the `user_custom_fields` model for updating `user_groups`?

File: api/models/user_custom_fields.py [hunk 2]
```diff
@@ -8,14 +8,38 @@ class UserCustomFieldsQuery(BaseQuery):
...
+        self.filter(and_(UserCustomFields.user_phone == phone)).update(
+            {UserCustomFields.status: UserCustomFields.UserCustomFieldStatus.INACTIVE}
+        )
+
+    def set_custom_field_as_active(self, field_name, field_value, phone):
```
Comment (Satendra-SR) [inline api/models/user_custom_fields.py:16]: Why we are using different terminology for activating and deactivating scenarios? Check the inactive method naming convention above.

File: api/models/user_custom_fields.py [hunk 3]
```diff
@@ -8,14 +8,38 @@ class UserCustomFieldsQuery(BaseQuery):
...
+
     id = db.Column(db.Integer, primary_key=True)
     user_id = db.Column(db.Integer, db.ForeignKey("user.id"))
     registration_id = db.Column(db.Integer, db.ForeignKey("registration.id"))
     user_phone = db.Column(db.String(50), nullable=False)
     flow_run_uuid = db.Column(db.String(255))
     field_name = db.Column(db.String(255), nullable=False)
     field_value = db.Column(db.String(500), nullable=False)
+    status = db.Column(
+        db.String(100), nullable=False, default=UserCustomFieldStatus.INACTIVE
```
Comment (Satendra-SR) [inline api/models/user_custom_fields.py:44]: Any reason to mark it default Inactive?

File: api/services/user_contact_service.py [hunk 4]
```diff
@@ -148,3 +159,7 @@ def custom_fields_contidions(self, field_name, field_value):
     def mark_user_groups_as_inactive(self):
         models.UserGroup.query.mark_user_groups_as_inactive(self.user_phone)
         db.session.commit()
+
+    def mark_user_custom_fields_as_inactive(self):
+        models.UserCustomFields.query.mark_user_groups_as_inactive(self.user_phone)
```
Comment (Satendra-SR) [inline api/services/user_contact_service.py:164]: The function name and the action seem confusing to me. Could you explain the reason?

**Review by Satendra-SR — CHANGES_REQUESTED** [review]
A few things need clairity. @GauravGusain98 have a look

File: api/models/user_custom_fields.py (hunk 3 above)
Comment (GauravGusain98) [reply api/models/user_custom_fields.py:44]: No such reason. It's just that marking active is not feeling right as all the old data will then be marked as active until we receive the webhook call for a number.

File: api/models/user_custom_fields.py (hunk 2 above)
Comment (GauravGusain98) [reply api/models/user_custom_fields.py:16]: Corrected it.

File: api/services/user_contact_service.py (hunk 4 above)
Comment (GauravGusain98) [reply api/services/user_contact_service.py]: corrected it

File: api/models/user_custom_fields.py (hunk 1 above)
Comment (GauravGusain98) [reply api/models/user_custom_fields.py]: Missed updating the name of the function. I'll correct it.

**Review by Satendra-SR — APPROVED** [review]
Looks good.

**Pull request merged**